test.js

```diff
@@ -670,6 +670,108 @@ function buildTests (test, sync) {
       }, 0)
     })
   })
+
+  test('mode', (t) => {
```

Comment:
To test I looked at the current [nodejs tests for fs.open](https://github.com/nodejs/node/blob/master/test/parallel/test-fs-open-mode-mask.js).

Unfortunately those tests do not test extensively passing different modes.